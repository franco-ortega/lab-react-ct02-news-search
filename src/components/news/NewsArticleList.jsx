import React from 'react';
import PropTypes from 'prop-types';
import NewsArticle from './NewsArticle';

const NewsArticleList = ({ newsArticles }) => {
  const articleElements = newsArticles.map(newsArticle => (
    <li key={newsArticle.title + Date.now()}>
      <NewsArticle {...newsArticle}/>
    </li>
  ));

  return (
    <ul>
      {articleElements}
    </ul>
  );
};

NewsArticleList.propTypes = {
  newsArticles: PropTypes.arrayOf(PropTypes.shape({
    title: PropTypes.string.isRequired,
    author: PropTypes.string,
    description: PropTypes.string
  })).isRequired
};

export default NewsArticleList;
