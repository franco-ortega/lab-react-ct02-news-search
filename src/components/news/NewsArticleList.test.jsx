import React from 'react';
import { render, cleanup } from '@testing-library/react';
import NewsArticleList from './NewsArticleList';
// global.fetch = require('node-fetch');

describe('NewsArticleList component', () => {
  afterEach(() => cleanup());
  it('renders NewsArticleList', () => {
    const { asFragment } = render(<NewsArticleList newsArticles={[]} />);
    expect(asFragment()).toMatchSnapshot();
  });

});

