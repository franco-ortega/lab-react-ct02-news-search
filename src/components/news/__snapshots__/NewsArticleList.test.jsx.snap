// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NewsArticleList component renders NewsArticleList 1`] = `
<DocumentFragment>
  <ul
    data-testid="display"
  />
</DocumentFragment>
`;
